depthai_helpers/managers.py

```diff
@@ -629,10 +629,35 @@ def create_default_queues(self, device):
         for xin in filter(lambda node: isinstance(node, dai.node.XLinkIn), vars(self.nodes).values()):
             device.getInputQueue(xin.getStreamName(), maxSize=1, blocking=False)
 
+    def __calc_encodeable_size(self, source_size):
+        w, h = source_size
+        if w % 16 > 0:
+            new_w = w - (w % 16)
+            h = int((new_w / w) * h)
+            w = int(new_w)
+        if h % 2 > 0:
+            h -= 1
+        return w, h
+
     def mjpeg_link(self, node, xout, node_output):
         print("Creating MJPEG link for {} node and {} xlink stream...".format(node.getName(), xout.getStreamName()))
         videnc = self.p.createVideoEncoder()
-        if isinstance(node, dai.node.ColorCamera) or isinstance(node, dai.node.MonoCamera):
+        if isinstance(node, dai.node.ColorCamera):
+            if node.video == node_output:
+                size = self.__calc_encodeable_size(node.getVideoSize())
+                node.setVideoSize(size)
+                videnc.setDefaultProfilePreset(*size, node.getFps(), dai.VideoEncoderProperties.Profile.MJPEG)
+            elif node.preview == node_output:
```

Comment:
Just a note here: only `video` and `still` from ColorCamera can be encoded, as VideoEncoder needs NV12 format (or grayscale). But maybe at some point we could add an option for ColorCamera preview to output grayscale, then it might be usable (with planar only, and VideoEncoder taking the first plane only if the format is RGB).

Reply:
thanks! would it be possible to extend the supported formats? This would allow to encode not only preview stream but also passthrough of depth